be/src/exprs/agg/ols.h

```diff
@@ -0,0 +1,841 @@
+                        "xx_weighted.size2({}) == biased_size({}): check failed.",
+                        xx_weighted_inv.size1(), xx_weighted_inv.size2(), xx_weighted.size1(), xx_weighted.size2(),
+                        biased_size);
+            }
+            ublas::matrix<double> tmp =
+                    prod(static_cast<ublas::matrix<double>>(prod(xx_weighted_inv, xx_weighted)), xx_weighted_inv);
+            for (size_t i = 0; i < biased_size; ++i) {
+                std(0, i) = tmp(i, i);
+            }
+        }
+
+        std::transform(std.data().begin(), std.data().end(), std.data().begin(), [](double x) { return std::sqrt(x); });
+
+        double var_y = _stats.var_y();
+
+        ublas::matrix<double> t_stat(1, biased_size);
+        for (size_t i = 0; i < coef.size1(); ++i) {
+            t_stat(0, i) = coef(i, 0) / std(0, i);
+        }
+
+        ublas::matrix<double> p_value(1, biased_size);
+        auto student = boost::math::students_t_distribution<double>(df);
+        for (size_t i = 0; i < biased_size; ++i) {
+            double t_stat_elem = t_stat(0, i);
+            if (std::isnan(t_stat_elem)) {
+                p_value(0, i) = t_stat_elem;
+            } else if (std::isinf(t_stat_elem)) {
+                p_value(0, i) = 0;
+            } else {
+                p_value(0, i) = 2 * cdf(complement(student, std::abs(t_stat(0, i))));
+            }
+        }
+
+        double rows_required = var_predict_y(0, 0) / var_y;
+        double adjusted_rows_required = 1 - (1 - rows_required) * (n - 1) / df;
+
+        double sse = var_y - var_predict_y(0, 0);
+        double f_statistic = (var_predict_y(0, 0) / k) / (sse / df);
+        double f_value = 1;
+        if (f_statistic > 0) {
+            if (std::isnan(f_statistic)) {
+                f_value = f_statistic;
+            } else if (std::isinf(f_statistic)) {
+                f_value = 0;
+            } else {
+                f_value = cdf(complement(boost::math::fisher_f(k, df), f_statistic));
+            }
+        }
+
+        result.coef = std::move(coef);
+        result.std = std::move(std);
+        result.t_stat = std::move(t_stat);
+        result.p_value = std::move(p_value);
+        result.var_y = var_y;
+        result.n = n;
+        result.k = k;
+        result.df = df;
+        result.sigma = sigma;
+        result.standard_error = standard_error;
+        result.rows_required = rows_required;
+        result.adjusted_rows_required = adjusted_rows_required;
+        result.f_statistic = f_statistic;
+        result.f_value = f_value;
+        result.biased_size = biased_size;
+
+        return std::nullopt;
+    }
+
+    std::string get_result() const {
+        StatsResult stats_result;
+        auto status = calc_stats_result(stats_result);
+        if (status.has_value()) {
+            return status.value();
+        }
+
+        std::string result;
+        result += "\nCall:\n  lm( formula = y ~";
```

Comment:
Are you sure those headers are part of the result? 

1. It may not be able to be further used as the input, do you  know other db systems how to handle this? have you survey any?
2. How about just return json format?